src/firebase.ts

```diff
@@ -78,4 +88,149 @@ const getDocById = async (coll: string, id: string) => {
     return JSON.stringify(doc, null, 2);
 }
 
-export { db, queryFirebase, getLocationDict, getDocById };
\ No newline at end of file
+const getRecipeDoc = async (id: string): Promise<FirebaseRecipe> => {
+    const q = query(
+        collection(db, FIRESTORE_COLLECTIONS.RECIPES),
+        where(documentId(), "==", id)
+    );
+    const querySnapshot = await getDocs(q);
+    const docs: Array<FirebaseRecipe> = querySnapshot.docs.map((doc) => ({
```

Comment:
We may need to include the required `name` and `dedup_hash` as they are quired so Typescript will be happy about it.